src/charm.py

```diff
@@ -283,11 +317,15 @@ def _handle_status_update_config(self, event: HookEvent) -> None:
 
         self.unit.status = MaintenanceStatus("Configuring the container")
 
+        self.oauth.update_client_config(client_config=self._oauth_client_config)
+
         # Make sure the directory for the logfile exists
         if not self._container.isdir(str(LOG_DIR)):
             self._container.make_dir(path=str(LOG_DIR), make_parents=True)
             logger.info(f"Created directory {LOG_DIR}")
 
+        self._push_ca_certs()
```

Comment:
Just a **NONBLOCKING** question: why do we push certs here again? The certs have been taken care of in handling the `CertificateAvailableEvent`, right?

Reply:
This serves mostly as a sanity check, not sure if it is needed tbh. I would expect that you are correct and that is handled by the `CertificateAvailableEvent`, but there are cases where juju can be unpredictable. 

For example if the relation is removed, the data may still be there. By including this push in the "reconcilation" function, we ensure that the certificates will eventually be removed. I can try to remove that for now and we can add it again if we get errors

Reply:
ic~ no worries. We can keep it there.